addons/binding/org.openhab.binding.gpstracker/src/main/java/org/openhab/binding/gpstracker/internal/profile/GPSTrackerProfileFactory.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.gpstracker.internal.profile;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.CoreItemFactory;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.profiles.*;
+import org.eclipse.smarthome.core.thing.type.ChannelType;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.openhab.binding.gpstracker.internal.GPSTrackerBindingConstants;
+import org.osgi.service.component.annotations.Component;
+
+import java.util.Collection;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.openhab.binding.gpstracker.internal.GPSTrackerBindingConstants.CHANNEL_TYPE_DISTANCE;
+import static org.openhab.binding.gpstracker.internal.GPSTrackerBindingConstants.CHANNEL_TYPE_REGION;
+
+/**
+ * The {@link GPSTrackerProfileFactory} class defines and provides switch profile and its type of this binding.
+ *
+ * @author Gabor Bicskei - Initial contribution
+ */
+@NonNullByDefault
+@Component
+public class GPSTrackerProfileFactory implements ProfileFactory, ProfileAdvisor, ProfileTypeProvider {
+    /**
+     * Profile UID for trigger events
+     */
+    static final ProfileTypeUID UID_TRIGGER_SWITCH = new ProfileTypeUID(GPSTrackerBindingConstants.BINDING_ID, "triggerPresenceSwitch");
+
+    /**
+     * Profile UID for distance calculation
+     */
+    static final ProfileTypeUID UID_DISTANCE_SWITCH = new ProfileTypeUID(GPSTrackerBindingConstants.BINDING_ID, "distancePresenceSwitch");
+
+    /**
+     * Profile type for trigger events
+     */
+    private static final TriggerProfileType TRIGGER_SWITCH_TYPE = ProfileTypeBuilder.newTrigger(UID_TRIGGER_SWITCH, "Presence Switch for Trigger")
```

Comment:
My problem here was the translation question. Is it possible to translate these labels? BTW why you want me to delete the gpstracker_en.properties file?

Reply:
Translation shouldn't matter here - finding a good translation is up to the people localising it.

> BTW why you want me to delete the gpstracker_en.properties file?

Why do you want to keep it? You are duplicating the english labels (and you have to make sure to keep them in sync). property files make sense if you want to provide a language other than english.

Reply:
Removed. Personally I thing it is useful to have a language file with all the label keys need to be translated. It helps the translation to other languages...

Reply:
That's true, but it could be simply an example file with just the keys - I simply think it is not good to have the Engish texts maintained at two different places and requiring them to be in sync.